File: 3379. transformed-array.py
from typing import List
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        return [nums[(i + x) % n] for i, x in enumerate(nums)]


if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [3, -2, 1, 1]
    print(sol.constructTransformedArray(nums1))  # Expected: [1, 1, 1, 3]
    
    # Example 2
    nums2 = [-1, 4, -1]
    print(sol.constructTransformedArray(nums2))  # Expected: [-1, -1, 4]

File: 3379. transformed-array.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            result[i] = nums[(i + nums[i] % n + n) % n];
        }
        return result;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {3, -2, 1, 1};
        int[] result1 = sol.constructTransformedArray(nums1);
        System.out.print("[");
        for (int i = 0; i < result1.length; i++) {
            System.out.print(result1[i]);
            if (i < result1.length - 1) System.out.print(", ");
        }
        System.out.println("]");  // Expected: [1, 1, 1, 3]
        
        // Example 2
        int[] nums2 = {-1, 4, -1};
        int[] result2 = sol.constructTransformedArray(nums2);
        System.out.print("[");
        for (int i = 0; i < result2.length; i++) {
            System.out.print(result2[i]);
            if (i < result2.length - 1) System.out.print(", ");
        }
        System.out.println("]");  // Expected: [-1, -1, 4]
    }
}
